src/Cpp/1-getting-started/14-HelloCOM/Application.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <string_view>
+#include <cstdint>
+
+struct GLFWwindow;
+class Application
+{
+public:
+    Application(const std::string_view title);
+    virtual ~Application();
+
+    void Run();
+protected:
+    virtual bool Initialize();
+    virtual void Cleanup();
+    virtual void Render() = 0;
+    virtual void Update() = 0;
+
+    [[nodiscard]] GLFWwindow* GetWindow() const;
```

Comment:
I am not a fan of this formatting style tbh. BUT dont worry, leave it as is for now, we are working on `.clangformat`